File: app/src/main/java/cm/aptoide/pt/download/Md5Comparator.java
package cm.aptoide.pt.download;

import android.text.TextUtils;
import cm.aptoide.pt.utils.AptoideUtils;
import java.io.File;

public class Md5Comparator {

  private String generalFolderPath;

  public Md5Comparator(String generalFolderPath) {
    this.generalFolderPath = generalFolderPath;
  }

  public boolean compareMd5(String md5, String fileName) {
    return TextUtils.equals(
        AptoideUtils.AlgorithmU.computeMd5(new File(generalFolderPath + fileName)), md5);
  }
}

File: aptoide-installer/src/main/java/cm/aptoide/pt/aptoide_installer/data/download/filedownloader/Md5Comparator.kt
package cm.aptoide.pt.aptoide_installer.data.download.filedownloader

import android.text.TextUtils
import cm.aptoide.pt.utils.AptoideUtils
import java.io.File

class Md5Comparator(private val generalFolderPath: String) {

  fun compareMd5(md5: String, fileName: String): Boolean {
    return TextUtils.equals(
      AptoideUtils.AlgorithmU.computeMd5(File(generalFolderPath + fileName)), md5
    )
  }
}